### ISSUE TITLE
Bang functions

### ISSUE THREAD

**Issue by benwilson512** [CargoSense/ex_aws#28]
In Dynamo and other APIs, there is a clear value to having ! function corollaries to a lot of regular functions. The question is, for something like get_item! should it unwrap `{:ok, result}` to just `result` or `{:ok, %{"Item" => item}` to `item`. It's worth keeping in mind that if we do the latter it invalidates certain options like `:return_consumed_capacity` which add an attribute to the result map.

Basically the question boils down to what the ! asserts in something like scan! query! get_records! etc. Are we asserting that the requests succeeds in an HTTP sense or are we trying to have it do more? I'm presently inclined towards the former option, but will think it over more.

**Comment by benwilson512** [comment]
This is going to be handled in the generated branch.
